src/main/java/org/craftercms/studio/impl/v2/service/webdav/WebDavServiceImpl.java

```diff
@@ -89,7 +96,11 @@ public List<WebDavItem> list(@ValidateStringParam(name = "siteId")
                                  @ProtectedResourceId("siteId") final String siteId,
                                  @ValidateStringParam(name = "profileId") final String profileId,
                                  @ValidateStringParam(name = "path") final String path,
-                                 @ValidateStringParam(name = "type") final String type) throws WebDavException {
+                                 @ValidateStringParam(name = "type") final String type) throws WebDavException, SiteNotFoundException {
+        if (!siteService.exists(siteId)) {
```

Comment:
Let's use 
```
siteService.checkSiteExists(siteId)
```

Reply:
Fixed this. I also updated the PR to add `ConfigurationProfileNotFoundException`. Please find the detail [here](https://github.com/craftercms/craftercms/issues/5798#issuecomment-1367190473).